.github/workflows/ci.yml

```diff
@@ -0,0 +1,102 @@
+name: Docker Image CI
+
+on:
+    push:
+        branches:
+            - main
+    pull_request:
+        branches:
+            - main
+
+jobs:
+    mindmapper-backend-build:
+        runs-on: ubuntu-latest
+
+        steps:
+            - uses: actions/checkout@v2
+
+            - name: Set up QEMU
+              uses: docker/setup-qemu-action@v1
+
+            - name: Set up Docker Buildx
+              uses: docker/setup-buildx-action@v1
+
+            - name: Login to GitHub Container Registry
+              uses: docker/login-action@v1
+              with:
+                  registry: ghcr.io
+                  username: ${{ github.repository_owner }}
+                  password: ${{ secrets.GITHUB_TOKEN }}
+
+            - name: Build and export to Docker
+              uses: docker/build-push-action@v2
+              with:
+                  context: .
+                  target: development
+                  push: true
+                  tags: |
+                      ghcr.io/b310-digital/mindmapper/ci:${{ github.sha }}-dev
```

Comment:
this image is now shown in the package section. Unfortunately, it is named only "mindmapper" and only by looking at the URL you can see, that it is actually the dev image. I am also wondering how this will work on branches or is this only triggered when deployed to main?